### ISSUE TITLE
Windows Support?

### ISSUE THREAD

**Issue by TechStudent10** [ArthurSonzogni/json-tui#11]
Can I use this with Windows?

**Comment by ArthurSonzogni** [comment]
Not yet.

However, it shouldn't be that hard. We need to remove those lines:
https://github.com/ArthurSonzogni/json-tui/blob/5a223aab989d7d46cd59e68c3c2dec7b7eea1168/src/main.cpp#L305-L307

It is to support:
```cpp
program | json-tui
```
use case. We read first from the stdin which is initially piped connected to `program` output. After that, we read from the console.

I believe this is mostly a Unix concept. Do you know if something like this can be done on Windows? Maybe I can make Windows to skip this feature.

**Comment by TechStudent10** [comment]
@ArthurSonzogni I do not understand 90% of that. I'm not a C++ Developer so I'm not sure. The only thing I got was that:

> Not yet.
> 
> However, it shouldn't be that hard. We need to remove those lines:
> 
> https://github.com/ArthurSonzogni/json-tui/blob/5a223aab989d7d46cd59e68c3c2dec7b7eea1168/src/main.cpp#L305-L307
> 
> It is to support:
> 
> ```c++
> program | json-tui
> ```
> 
> use case. We read first from the stdin which is initially piped connected to `program` output. After that, we read from the console.

is a Unix concept and that something like this won't work on a WinNT system.

**Comment by ArthurSonzogni** [comment]
>  do not understand 90% of that.

TLDR:  I can make it work Windows, one way or the other. I will make something soon. Today or tomorrow.

**Comment by TechStudent10** [comment]
@ArthurSonzogni oh okay! Thanks!

**Comment by ArthurSonzogni** [comment]
Hi @TechStudent11 
Windows should now be supported! Could you please take a look and tell me if there are any problems?

I don't have a WIndows OS, so I can't really test it myself. However the bot was able to build it a probduce a release:
https://github.com/ArthurSonzogni/json-tui/releases/tag/v1.2.0

**Comment by TechStudent10** [comment]
@ArthurSonzogni Woohoo! I'll test it out right now!

**Comment by TechStudent10** [comment]
Proud to say that it works:
![image](https://user-images.githubusercontent.com/76978184/149243506-8c25861c-d3d8-4a38-8d6a-67b71fd2af74.png)

**Comment by ArthurSonzogni** [comment]
Awesome! Thank you very much for checking!

**Comment by TechStudent10** [comment]
> Awesome! Thank you very much for checking!

No problem!

**Comment by TechStudent10** [comment]
The fact that I just came back and now almost understand all of this:

> Not yet.
> 
> However, it shouldn't be that hard. We need to remove those lines:
> 
> https://github.com/ArthurSonzogni/json-tui/blob/5a223aab989d7d46cd59e68c3c2dec7b7eea1168/src/main.cpp#L305-L307
> 
> It is to support:
> 
> ```c++
> program | json-tui
> ```
> 
> use case. We read first from the stdin which is initially piped connected to `program` output. After that, we read from the console.
> 
> I believe this is mostly a Unix concept. Do you know if something like this can be done on Windows? Maybe I can make Windows to skip this feature.
